sql/modules/Roles.sql

```diff
@@ -356,18 +421,75 @@ SELECT lsmb__grant_perms('contact_read', 'file_entity', 'SELECT');
-SELECT lsmb__create_role('contact_class_referral');
-SELECT lsmb__create_role('contact_class_lead');
-SELECT lsmb__create_role('contact_class_hot_lead');
-SELECT lsmb__create_role('contact_class_cold_lead');
-SELECT lsmb__create_role('contact_class_sub_contractor');
-SELECT lsmb__create_role('contact_class_robot');
-
-SELECT lsmb__create_role('contact_create');
+SELECT lsmb__create_role('contact_class_vendor',
+                         $DOC$
+                         This role allows access to vendor contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_customer',
+                         $DOC$
+                         This role allows access to customer contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_employee',
+                         $DOC$
+                         This role allows access to employee contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_contact',
+                         $DOC$
+                         This role allows access to contact data (e-mail, phone, etc) of all kinds of contacts (customer/vendor/...).
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_referral',
+                         $DOC$
+                         This role allows access to referral contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_lead',
+                         $DOC$
+                         This role allows access to sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_hot_lead',
+                         $DOC$
+                         This role allows access to hot sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_cold_lead',
+                         $DOC$
+                         This role allows access to cold sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
```

Comment:
aren't all `contact_class_*` roles missing the `contact_delete` option?

Reply:
yep. they are.

Reply:
Adjusted.